internal/provider/resource_switch_port_settings.go

```
@@ -0,0 +1,1304 @@
+
+	// Map response body to schema
+	state.ID = types.StringValue(settings.Id)
+	state.Name = types.StringValue(string(settings.Name))
+	state.Description = types.StringValue(settings.Description)
+	state.TimeCreated = types.StringValue(settings.TimeCreated.String())
+	state.TimeModified = types.StringValue(settings.TimeModified.String())
+
+	model, diags := toTerraformModel(settings)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	state.Addresses = model.Addresses
+	state.BGPPeers = model.BGPPeers
+	state.Groups = model.Groups
+	state.Interfaces = model.Interfaces
+	state.Links = model.Links
+	state.PortConfig = model.PortConfig
+	state.Routes = model.Routes
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *switchPortSettingsResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) { // plan is the resource data model for the update request.
+	var plan switchPortSettingsModel
+	// state is the resource data model for the current state.
+	var state switchPortSettingsModel
+
+	// Read Terraform plan data into the plan model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read Terraform prior state data into the state model to retrieve ID
+	// which is a computed attribute, so it won't show up in the plan.
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateTimeout, diags := plan.Timeouts.Update(ctx, defaultTimeout())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, updateTimeout)
+	defer cancel()
+
+	params, _ := toOxideParams(plan)
```

Same here, let's return the errors